crates/common/src/state.rs

```diff
@@ -140,9 +140,10 @@ pub trait State {
         Ok(Some(id))
     }
 
-    fn get_sudt_balance(&self, sudt_id: u32, id: u32) -> Result<u128, Error> {
+    fn get_sudt_balance(&self, sudt_id: u32, short_address: &[u8]) -> Result<u128, Error> {
+        assert_eq!(short_address.len(), 20);
         // get balance
-        let balance = self.get_value(sudt_id, &H256::from_u32(id))?;
+        let balance = self.get_raw(&build_account_key(sudt_id, short_address))?;
```

Comment:
I think the `addr_len` is meaningless, since the information is include in `short_address`. 

For example, for same content but different length short address:
```
short_address4 = 0x00000038
short_address6 = 0x000000000038
```
They will produce different key with `build_account_key`.

Reply:
A type flag may a good idea, will add it.

Reply:
> I think the `addr_len` is meaningless, since the information is include in `short_address`.
> 
> For example, for same content but different length short address:
> 
> ```
> short_address4 = 0x00000038
> short_address6 = 0x000000000038
> ```
> 
> They will produce different key with `build_account_key`.

You are right about the hashing result, but what I want to do is to provide a standard way to serialize the args of build_account_key, since the args is variable now, to pre-append length on each args can prevent users construct a collision result, for example:

If we construct a key with two variable args:

```
var1: "abc" | var2: "de"

and

var1: "ab" | var2: "cde"

produce the same hashing result.

after pre-append the length before arg:

var1: "3abc" | var2: "2de"

var1: "2ab" | var2: "3cde"

produce the different result.
```